src/main/java/jsr223/nativeshell/NativeShellRunner.java

```diff
@@ -274,12 +271,15 @@ private void addArrayBindingAsEnvironmentVariable(String bindingKey, Object[] bi
         }
     }
 
-    private static Thread readProcessOutput(final InputStream processOutput, final Writer contextWriter) {
+    private static Thread readProcessOutput(final InputStream processOutput, final Writer contextWriter,
+            StringBuilder captureOutput) {
         return new Thread(new Runnable() {
             @Override
             public void run() {
                 try {
-                    pipe(new BufferedReader(new InputStreamReader(processOutput)), new BufferedWriter(contextWriter));
+                    pipe(new BufferedReader(new InputStreamReader(processOutput)),
```

Comment:
I am sorry but I think you need to put BufferedReader and InputStreamReader objects into try-with-resources.

Reply:
Ok